src/Apps/Legion/Models/Context/WizardNavigationContext/WizardNavigationContext.tsx

```diff
@@ -0,0 +1,83 @@
+import produce from 'immer';
+import React, { useContext, useReducer } from 'react';
+import { getDebugLogger } from '../../../../../Models/Services/Utils';
+import {
+    IWizardNavigationContext,
+    IWizardNavigationContextProviderProps,
+    IWizardNavigationContextState,
+    WizardNavigationContextAction,
+    WizardNavigationContextActionType
+} from './WizardNavigationContext.types';
+
+export const WizardNavigationContext = React.createContext<IWizardNavigationContext>(
+    null
+);
+export const useWizardNavigationContext = () =>
+    useContext(WizardNavigationContext);
+
+const debugLogging = false;
+const logDebugConsole = getDebugLogger('NavigationContext', debugLogging);
+
+export const NavigationContextReducer: (
```

Comment:
do we need context for doing the navigation? Is this so that each page can trigger a transition to the next step?

Reply:
Yeah, it is to be able to manage step transitions outside the Stepper component.